src/main/kotlin/apply/application/ApplicantService.kt

```diff
@@ -3,21 +3,39 @@ package apply.application
 import apply.domain.applicant.Applicant
 import apply.domain.applicant.ApplicantRepository
 import apply.domain.applicant.Gender
+import apply.domain.applicant.dto.ApplicantRequest
+import apply.security.JwtTokenProvider
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import support.createLocalDate
 import javax.annotation.PostConstruct
 
 @Transactional
 @Service
-class ApplicantService(private val applicantRepository: ApplicantRepository) {
+class ApplicantService(
+    private val applicantRepository: ApplicantRepository,
+    private val jwtTokenProvider: JwtTokenProvider
+) {
     fun findAll(): List<Applicant> {
         return applicantRepository.findAll()
     }
 
     fun findByValue(value: String): List<Applicant> =
         applicantRepository.findByNameContainingOrEmailContaining(value, value)
 
+    fun validateOrCreateApplicantAndGenerateToken(applicantRequest: ApplicantRequest): String {
+        applicantRepository.findByEmail(applicantRequest.email)
+            ?.validate(applicantRequest)
+            ?: createApplicant(applicantRequest)
+
+        return jwtTokenProvider.createToken(applicantRequest.email)
+    }
+
+    private fun createApplicant(applicantRequest: ApplicantRequest) {
+        val newApplicant = applicantRequest.toEntity()
+        applicantRepository.save(newApplicant)
```

Comment:
재사용하지 않는 지역 변수의 수는 최소화하면 어떨까요?

```suggestion
        applicantRepository.save(applicantRequest.toEntity())
```